simple_scheduler/jobs/ecs_job.py

```diff
@@ -0,0 +1,122 @@
+"""A sample job that prints string."""
+
+import time
+import logging
+from ndscheduler import job
+
+logger = logging.getLogger(__name__)
+
+try:
+    import boto3
+
+    client = boto3.client('ecs')
+except ImportError:
+    logger.warning('boto3 is not installed. ECSTasks require boto3')
+
+POLL_TIME = 2
+
+
+class ECSJob(job.JobBase):
+    @classmethod
+    def meta_info(cls):
+        return {
+            'job_class_string': '%s.%s' % (cls.__module__, cls.__name__),
+            'notes': 'This will execute a AWS ECS RunTask!',
+            'arguments': [
+                {'type': 'string', 'description': 'ECS Cluster to run on'},
+                {'type': 'string', 'description': 'task_def_arn'},
+                {'type': 'array[dict]', 'description': 'task_def'},
+                {'type': 'string', 'description': 'Directly corresponds to the '
+                                                  '`overrides` parameter of runTask API'}
+            ],
+            'example_arguments': '["ClusterName", None, "arn:aws:ecs:<region>'
+                                 ':<user_id>:task-definition/<family>:<tag>", None]'
+        }
+
+    def _get_task_statuses(self, task_ids):
+        """
+        Retrieve task statuses from ECS API
+
+        Returns list of {RUNNING|PENDING|STOPPED} for each id in task_ids
+        """
+        logger.debug('Get status of task_ids: {}'.format(task_ids))
+        response = client.describe_tasks(tasks=task_ids, cluster=self.cluster)
+
+        # Error checking
+        if response['failures']:
+            raise Exception('There were some failures:\n{0}'.format(
```

Comment:
Don't raise a generic exception. Make a custom exception that subclasses Exception, e.g., ECSJobException.